pylti/common.py

```diff
@@ -350,7 +344,91 @@ def generate_request_xml(message_identifier_id, operation,
         text_string = etree.SubElement(result_score, 'textString')
         text_string.text = score.__str__()
     ret = "<?xml version='1.0' encoding='utf-8'?>\n{}".format(
-        etree.tostring(root, encoding='utf-8'))
+        etree.tostring(root, encoding='utf-8').decode('utf-8'))
 
     log.debug("XML Response: \n%s", ret)
     return ret
+
+
+class SignatureMethod_HMAC_SHA1_Unicode(oauth2.SignatureMethod_HMAC_SHA1):
+    """
+    Temporary workaround for
+    https://github.com/joestump/python-oauth2/issues/207
+    """
+
+    def check(self, request, consumer, token, signature):
+        """Returns whether the given signature is the correct signature for
+        the given consumer and token signing the given request."""
+        built = self.sign(request, consumer, token)
+        if isinstance(signature, STRING_TYPES):
+            signature = signature.encode("utf8")
+        return built == signature
+
+
+class SignatureMethod_PLAINTEXT_Unicode(oauth2.SignatureMethod_PLAINTEXT):
+    """
+    Temporary workaround for
+    https://github.com/joestump/python-oauth2/issues/207
+    """
+
+    def check(self, request, consumer, token, signature):
+        """Returns whether the given signature is the correct signature for
+        the given consumer and token signing the given request."""
+        built = self.sign(request, consumer, token)
+        if isinstance(signature, STRING_TYPES):
+            signature = signature.encode("utf8")
+        return built == signature
+
+
+class Request_Fix_Duplicate(oauth2.Request):
+    """
+    Temporary workaround for
+    https://github.com/joestump/python-oauth2/pull/197
+    """
+
+    def get_normalized_parameters(self):
+        """Return a string that contains the parameters that must be signed."""
+        items = []
+        for key, value in self.items():
+            if key == 'oauth_signature':
+                continue
+            # 1.0a/9.1.1 states that kvp must be sorted by key, then by value,
+            # so we unpack sequence values into multiple items for sorting.
+            if isinstance(value, STRING_TYPES):
+                items.append((oauth2.to_utf8_if_string(key),
+                              oauth2.to_utf8(value)))
+            else:
+                try:
+                    value = list(value)
+                except TypeError as e:
+                    assert 'is not iterable' in str(e)
```

Comment:
This code is a slight adaptation of the method with the same name in python-oauth2. It is in fact the function patched with https://github.com/joestump/python-oauth2/pull/197.

The assert is part of the original code.

I forgot to add the specific copyright notice; I will do it with the remaining corrections :-)